docs/src/main/asciidoc/mongodb-panache-guide.adoc

```diff
@@ -0,0 +1,505 @@
+
+[source,java]
+--
+List<Person> persons = Person.list(Sort.by("name").and("birth"));
+
+// and with more restrictions
+List<Person> persons = Person.list("status", Sort.by("name").and("birth"), Status.Alive);
+--
+
+The `Sort` class has plenty of methods for adding columns and specifying sort direction.
+
+== Adding entity methods
+
+In general, we recommend not adding custom queries for your entities outside of the entities themselves,
+to keep all model queries close to the models they operate on. So we recommend adding them as static methods
+in your entity class:
+
+[source,java]
+--
+@MongoEntity
+public class Person extends PanacheMongoEntity {
+    public String name;
+    public LocalDate birth;
+    public Status status;
+
+    public static Person findByName(String name){
+      return find("name", name).firstResult();
+    }
+
+    public static List<Person> findAlive(){
+      return list("status", Status.Alive);
+    }
+
+    public static void deleteStefs(){
+      delete("name", "Stef");
+    }
+}
+--
+
+== Simplified queries
+
+Normally, MongoDB queries are of this form: `{'firstname': 'John', 'lastname':'Doe'}`, this is what we called native MongoDB queries.
+
+You can use them if you want, but we also support what we called **PanacheQL** that can be seen as a subset of JP-QL (or HQL) and allow to easily express a query. MongoDB Panache will then map it to native MongoDB query.
+
+If your query does not start with `{`, we will imply PanacheQL query
+
+- `<singleColumnName>` (and single parameter) which will expand to `{'singleColumnName': '?'}`
+- `<query>` will expand to `{<query>}` where we will map the PanacheQL query to MongoDB native query form. We support the following operators that will be mapped to the corresponding MongoDB operators: 'and', 'or' ( mixing 'and' and 'or' is not currently supported), '=', '>', '>=', '<', '<=', '!=', 'is null', 'is not null', and 'like' that is mapped to the MongoDB `$regex` operator.
+
+Here are some query examples:
+
+- `firstname = ?1 and status = ?2` will be mapped to `{'firstname': ?1, 'status': ?2}`
+- `amount > ?1 and firstname != ?2` will be mapped to `{'amount': {'$gt': ?1}, 'firstname': {'$ne': ?2}}`
+- `lastname like ?1` will be mapped to `{'lastname': {'$regex': ?1}}`. Be careful that this will be link:https://docs.mongodb.com/manual/reference/operator/query/regex/#op._S_regex[MongoDB regex] support and not SQL like pattern. 
+- `lastname is not null` will be mapped to `{'lastname':{'$exists': true}}`
+
+We will also handle some basic Date type transofmation: all fields of type `Date`, `LocalDate` or `LocalDateTime` will be mapped to MongoDB `ISODate` type.
+
+MongoDB Panache also support extented MongoDB queries by providing a `Document` query, this is supported by the find/list/stream/cout/delete methods.
+
+== Query parameters
+
+You can pass query parameters, for both native and PanacheQL queries, by index (1-based):
```

Comment:
`You can pass query parameters, for both native and PanacheQL queries, by index (1-based) as shown below:`

Reply:
done